src/NServiceBus.RavenDB.Tests/Infrastructure/ReusableDB.cs

```diff
@@ -0,0 +1,85 @@
+﻿namespace NServiceBus.RavenDB.Tests
+{
+    using System;
+    using System.Threading;
+    using Raven.Client;
+    using Raven.Client.Document;
+
+    class ReusableDB : IDisposable
+    {
+        private readonly string databaseName;
+        private readonly bool deleteOnCompletion;
+
+        public ReusableDB(bool deleteOnCompletion = true)
+        {
+            this.deleteOnCompletion = deleteOnCompletion;
+            databaseName = Guid.NewGuid().ToString("N");
+
+            Console.WriteLine($"Provisioned new Raven database name {databaseName}");
+        }
+
+        public IDocumentStore NewStore()
+        {
+            Console.WriteLine();
+            Console.WriteLine($"Creating new DocumentStore for {databaseName}");
+            return new DocumentStore
+            {
+                Url = "http://localhost:8081",
+                DefaultDatabase = databaseName
+            };
+        }
+
+        public void WaitForIndexing(IDocumentStore store)
+        {
+            while (store.DatabaseCommands.GetStatistics().StaleIndexes.Length != 0)
+            {
+                Thread.Sleep(250);
+            }
+
+            Thread.Sleep(100);
+        }
+
+        public void Dispose()
+        {
+            if (!deleteOnCompletion)
+            {
+                return;
+            }
+
+            var docStore = new DocumentStore
+            {
+                Url = "http://localhost:8081"
+            };
+
+            docStore.Initialize();
+
+            for (var i = 0; i < 4; i++)
```

Comment:
Might be nitpicking being a 'test' code and all but could we write this more functionally? The intent would be expressed better when not mixed with the retry logic. http://stackoverflow.com/questions/1563191/c-sharp-cleanest-way-to-write-retry-logic